pysys/process/user.py

```diff
@@ -426,19 +437,25 @@ def getDefaultEnvirons(self, command=None, **kwargs):
 			e['DYLD_LIBRARY_PATH'] = DYLD_LIBRARY_PATH
 				
 		if not IS_WINDOWS:
-			if getattr(PROJECT, 'defaultEnvironsDefaultLang',''):
-				e['LANG'] = PROJECT.defaultEnvironsDefaultLang
+			if getattr(self.project, 'defaultEnvironsDefaultLang',''):
+				e['LANG'] = self.project.defaultEnvironsDefaultLang
 		
 		if command == sys.executable:
-			# ensure it's possible to run another instance of this Python
-			# (but only if full path exactly matches)
-			# keep it as clean as possible by not passing sys.path/PYTHONPATH
+			# Ensure it's possible to run another instance of this Python, by adding it to the start of the path env vars
+			# (but only if full path to the Python executable exactly matches).
+			# Keep it as clean as possible by not passing sys.path/PYTHONPATH
+			# - but it seems we do need to copy the LD_LIBRARY_PATH from the parent process to ensure the required libraries are present.
+			# Do not set PYTHONHOME here, as doesn't work well in virtualenv, and messes up grandchildren 
+			# processes that need a different Python version
 			e['PATH'] = os.path.dirname(sys.executable)+os.pathsep+e['PATH']
-			e[LIBRARY_PATH_ENV_VAR] = os.getenv(LIBRARY_PATH_ENV_VAR,'')+os.pathsep+e[LIBRARY_PATH_ENV_VAR]
-			isvirtualenv = hasattr(sys, 'real_prefix') or (
-				hasattr(sys, 'base_prefix') and sys.base_prefix != sys.prefix)
-			if not isvirtualenv:
-				e['PYTHONHOME'] = sys.prefix
+
+			if LIBRARY_PATH_ENV_VAR != 'PATH': # if it's an os with something like LD_LIBRARY_PATH
+				# it's a shame this is necessary, but there's no sane way to unpick which libraries are 
+				# actually required on Unix. 
+				e[LIBRARY_PATH_ENV_VAR] = os.getenv(LIBRARY_PATH_ENV_VAR,'')+os.pathsep+e.get(LIBRARY_PATH_ENV_VAR,'')
```

Comment:
Again, is this going to cause anything weird to happen if LIBRARY_PATH_ENV_VAR doesn't exist in the parent environment?

Reply:
Resolved